### PR TITLE
[DEV 4622] Add support for now in expressions.

### PR DESCRIPTION
**Pull request by nonibansal**
> [!IMPORTANT]
> Add support for `now()` in expressions for time-based calculations, updating expression handling, serialization, and testing.
> 
>   - **Behavior**:
>     - Add `now()` function in `expr.py` for current time in expressions.
>     - Update `Expr` class to handle `Now` expressions.
>     - Modify `QueryEngine` to include `NOW_COL_NAME` in expression evaluation.
>   - **Testing**:
>     - Add `test_now()` in `test_expr.py` to validate `now()` functionality.
>     - Add integration test `test_now()` in `test_expr.py` to verify end-to-end behavior.
>   - **Serialization**:
>     - Update `ExprSerializer` in `serializer.py` to serialize `Now` expressions.
>     - Update `visitor.py` to handle `Now` expressions in `ExprPrinter` and `FetchReferences`.
>   - **Misc**:
>     - Update `CHANGELOG.md` for version 1.5.33.
>     - Bump version in `pyproject.toml` to 1.5.33.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=fennel-ai%2Fclient&utm_source=github&utm_medium=referral)<sup> for 880f94339fbdd8b6778d34fb3d9a1b619dd0c9ce. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: fennel/expr/expr.py [hunk 1]
```diff
@@ -27,6 +29,10 @@
 )
 
 
+class ExprContext(BaseModel):
```
Comment (nikhilgarg28) [inline fennel/expr/expr.py:32]: Rename to `EvalContext` please.

File: pyproject.toml [hunk 2]
```diff
@@ -43,20 +43,20 @@ pyyaml = "^6.0.1"
...
+build-backend = "maturin"
 
-# [tool.maturin]
-# name = "fennel_data_lib"
-# sdist-directory = "python_package"
-# manifest-path = "../server/fennel_data_lib/Cargo.toml"
+[tool.maturin]
+name = "fennel_data_lib"
+sdist-directory = "python_package"
+manifest-path = "../server/fennel_data_lib/Cargo.toml"
```
Comment (nikhilgarg28) [inline pyproject.toml:59]: I suppose you'd need to revert this before merging?

**Review by nikhilgarg28 — CHANGES_REQUESTED** [review]
Could you please also add this to docs?

File: fennel/expr/expr.py (hunk 1 above)
Comment (nonibansal) [reply fennel/expr/expr.py:32]: changed

File: pyproject.toml (hunk 2 above)
Comment (nonibansal) [reply pyproject.toml:59]: Yup.

File: fennel/expr/expr.py (hunk 1 above)
Comment (nonibansal) [reply fennel/expr/expr.py]: renamed

File: pyproject.toml (hunk 2 above)
Comment (nonibansal) [reply pyproject.toml]: yup reverted

File: docs/api.yml [hunk 3]
```diff
@@ -98,6 +98,7 @@ sidebar:
       - "api-reference/expressions/not"
       - "api-reference/expressions/typeof"
       - "api-reference/expressions/when"
+      - "api-reference/expressions/now"
```
Comment (nikhilgarg28) [inline docs/api.yml:101]: We keep all API reference tabs alphabetically sorted within a section. Please move it after `not`

Comment (nonibansal) [reply docs/api.yml]: changed

File: docs/pages/api-reference/expressions/now.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+title: Now
+order: 0
+status: published
+---
+### Now
+
+Function to now timestamp as a column, similar to `current_timestamp()` in Polars or Spark.
```
Comment (nikhilgarg28) [inline docs/pages/api-reference/expressions/now.md:8]: The grammar needs to be adjusted + better to reference `datetime.now` as an example. Maybe `Function to get the current timestamp, similar to what datetime.now does in Python`.

**Review by nikhilgarg28 — APPROVED** [review]
Approving tentatively though plz make the couple small changes suggested.

File: docs/pages/api-reference/expressions/now.md (hunk 4 above)
Comment (nonibansal) [reply docs/pages/api-reference/expressions/now.md:8]: changed

**Pull request merged**